pynamodb/models.py

```diff
@@ -1211,21 +1260,32 @@ def _serialize(self, attr_map=False, null_check=True):
             if serialized is None:
                 continue
             if attr_map:
-                attrs[attributes][attr.attr_name] = {
-                    ATTR_TYPE_MAP[attr.attr_type]: serialized
-                }
+                attrs[attributes][attr.attr_name] = Model._serialize_value(attr, value)
             else:
                 if attr.is_hash_key:
                     attrs[HASH] = serialized
                 elif attr.is_range_key:
                     attrs[RANGE] = serialized
                 else:
-                    attrs[attributes][attr.attr_name] = {
-                        ATTR_TYPE_MAP[attr.attr_type]: serialized
-                    }
+                    attrs[attributes][attr.attr_name] = Model._serialize_value(attr, value)
         return attrs
 
     @classmethod
+    def _serialize_value(cls, attr, value):
+        """
+        Serializes a value for use with DynamoDB
+
+        :param value: a value to serialize
+        """
+        serialized = attr.serialize(value)
```

Comment:
the passed value is already serialized on [1210](https://github.com/jlafon/PynamoDB/pull/194/files#diff-66d6c6d3d504b9e11a4e236e67133ddeL1210) right?

we should either remove that or skip this